internal/memorylimiter/memorylimiter.go

```diff
@@ -0,0 +1,296 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package memorylimiter // import "go.opentelemetry.io/collector/internal/memorylimiter"
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/iruntime"
+)
+
+const (
+	mibBytes = 1024 * 1024
+
+	// Minimum interval between forced GC when in soft limited mode. We don't want to
+	// do GCs too frequently since it is a CPU-heavy operation.
+	minGCIntervalWhenSoftLimited = 10 * time.Second
+)
+
+var (
+	// ErrDataRefused will be returned to callers of ConsumeTraceData to indicate
+	// that data is being refused due to high memory usage.
+	ErrDataRefused = errors.New("data refused due to high memory usage")
+
+	// Construction errors
+	errCheckIntervalOutOfRange = errors.New(
+		"checkInterval must be greater than zero")
+
+	errLimitOutOfRange = errors.New(
+		"memAllocLimit or memoryLimitPercentage must be greater than zero")
+
+	errMemSpikeLimitOutOfRange = errors.New(
+		"memSpikeLimit must be smaller than memAllocLimit")
+
+	errPercentageLimitOutOfRange = errors.New(
+		"memoryLimitPercentage and memorySpikePercentage must be greater than zero and less than or equal to hundred",
+	)
+
+	errShutdownNotStarted = errors.New("no existing monitoring routine is running")
+
+	// make it overridable by tests
+	GetMemoryFn = iruntime.TotalMemory
+)
+
+// Config defines configuration for memory memoryLimiter processor.
+type Config struct {
```

Comment:
Please move this to `config.go` similar to how it used to be

Reply:
moved as is